### PR TITLE
Adds other embargo reason.

### PR DESCRIPTION
**Pull request by straleyb**
Adds in text field for "Other embargo reason"

### CODE REVIEW

File: app/assets/javascripts/embargo_reason_field.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+$( document ).ready(function() {
+  $('#embargo_reason_select').change(function() {
```
Comment (revgum) [inline app/assets/javascripts/embargo_reason_field.js]: If the user leaves the default selection without triggering a "change" on the select, what would happen?

File: app/assets/javascripts/embargo_reason_field.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    var val = $("#embargo_reason_select option:selected").val();
+    if (val == 'Other...') {
+      $('#article_embargo_reason').removeClass('hidden');
```
Comment (revgum) [inline app/assets/javascripts/embargo_reason_field.js]: clear the value for in the input, allowing the user to enter their own.

File: app/assets/javascripts/embargo_reason_field.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    } else {
+      $('#article_embargo_reason').addClass('hidden');
+      $('#article_embargo_reason').val(val);
+    }
+  });
```
Comment (revgum) [inline app/assets/javascripts/embargo_reason_field.js]: I feel like when the embargo radio is selected, you would want to trigger "change" initially for the embargo_reason_select

File: app/assets/javascripts/embargo_reason_field.js (hunk 1 above)
Comment (straleyb) [inline app/assets/javascripts/embargo_reason_field.js]: So if they dont trigger a change, it accepts the form per usual. I didnt want to make the embargo reason required. What I can do though, if it sounds like a good idea, is to pre-fill the value for them and make not filling embargo reason impossible.

**Pull request merged**